### DISCUSSION TITLE
Markdown text with Multimedia

### DISCUSSION THREAD

**Discussion by ylc395** [Milkdown/milkdown#929] [Q&A]
Here is a fragment of Markdown:

```
![](https://url-of-a-resource.com/resource)
```

The URL may indicate an image/video/audio or resource of other type (maybe a PDF file). When parsing markdown, `remark` can not recognize the resource's mime type since we can not query it via an HTTP request (remark plugin can not do async task). So, If I want to render this online resource correctly, for example, using `<video>` element for video resource, What should I do?

**Comment by Saul-Mirone** [comment]
You can create a placeholder for your element and do the async request in background. Once you get the response, you can get back to the document and replace the placeholder with your resource. It'll be very similar to [`@milkdown/plugin-upload`](https://github.com/Milkdown/milkdown/blob/main/packages/plugin-upload/src/upload.ts#L50).

**Reply by ylc395** [reply]
Anyway, we should provide some new node schemas (like video node, audio node, etc) to Prosemirror. Is there any chance to omit `parseMarkdown` and `toMarkdown` when using `$nodeSchema` since a markdown syntax for video/audio/etc is not required in this case?

**Reply by Saul-Mirone** [reply]
Have you tried to extend the schema?

**Reply by ylc395** [reply]
maybe this will work

```js
    parseMarkdown: {
      match: () => false,
      runner: noop,
    },
    toMarkdown: {
      match: () => false,
      runner: noop,
    },

```
